lib/msal-browser/src/controllers/StandardController.ts

```diff
@@ -2386,6 +2389,72 @@ export class StandardController implements IController {
         );
         return res;
     }
+
+    private logMultipleInstances(
+        performanceEvent: InProgressPerformanceEvent,
+        request?: InitializeApplicationRequest
+    ): void {
+        const appId = request?.appId || this.config.auth.clientId;
+        // @ts-ignore
+        window.msal = window.msal || {};
+        // @ts-ignore
+        window.msal.appIds = window.msal.appIds || [];
+
+        // @ts-ignore
+        if (this.checkForSameClientId(appId, window.msal.appIds)) {
+            this.logger.warning(
+                "There is already an instance of MSAL.js in the window with the same client id."
+            );
+            // @ts-ignore
+        } else if (this.checkForMultipleInstances(appId, window.msal.appIds)) {
+            this.logger.warning(
+                "There is already an instance of MSAL.js in the window."
+            );
+        }
+
+        // @ts-ignore
+        if (!(window.msal.appIds.includes(appId) && appId.includes("."))) {
+            // @ts-ignore
+            window.msal.appIds.push(appId);
+        }
+
+        collectInstanceStats(appId, performanceEvent);
+    }
+
+    private checkForSameClientId(id: string, appIdArray: string[]): boolean {
+        if (id.includes(".")) {
+            const [clientId] = id.split(".");
+            /**
+             * Check for 1P applications.
+             * appId = clientId.channelId
```

Comment:
Are we only including `channelId` to workaround the double counting scenario or is this used for something else? If so I think there's a cleaner way we can handle this, let's chat.

Reply:
Not just double counting, also used for two embedded apps instances for the same clientid, which should be counted.